idea/meta-storage/src/services/meta/meta.service.ts

```diff
@@ -0,0 +1,51 @@
+import { plainToClass } from 'class-transformer';
+import { HexString } from '@polkadot/util/types';
+import { getProgramMetadata } from '@gear-js/api';
+
+import { CreateMetaInput } from './input/create-meta.input';
+import { Meta } from '../../database/entities';
+import { metaRepository } from '../../database/repositories';
+import { MetaAlreadyExists, MetadataNotFound } from '../../common/errors';
+import { generateHash } from '../../common/helpers/generate-hash';
+
+export const metaService = {
+  async createByIndexer(createMetaInput: CreateMetaInput): Promise<Meta | null> {
+    const metaDB = await metaRepository.get(createMetaInput.hash);
+
+    if (metaDB) return;
+
+    const metaTypeDB = plainToClass(Meta, {
```

Comment:
`plainToClass` is deprecated if I'm not wrong. `plaintToInstance` is okay

Reply:
Done